solutions/usaco-787.mdx

```diff
@@ -87,4 +86,81 @@ int main() {
 ```
 
 </CPPSection>
+<JavaSection>
+```java
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.StringTokenizer;
+
+public class rental {
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new FileReader("rental.in"));
+        PrintWriter out = new PrintWriter("rental.out");
+        StringTokenizer s = new StringTokenizer(br.readLine());
+        int n = Integer.parseInt(s.nextToken());
+        int m = Integer.parseInt(s.nextToken());
+        int r = Integer.parseInt(s.nextToken());
+        Integer [] production = new Integer [n];
+        for (int i = 0; i < n; i++) {
+            production[i] = Integer.parseInt(br.readLine());
+        }
+        Integer [][] shops = new Integer [m][2];
```

Comment:
Could you make a `Shop` class? I think that would make it easier to understand what property is what, and you can also put the comparator function in there.

Reply:
Aight